sbndaq-artdaq/Generators/Common/CAENV1730Readout_generator.cc

```diff
@@ -1332,19 +1332,73 @@ bool sbndaq::CAENV1730Readout::checkHWStatus_(){
     memfullStream << "CAENV1730.Card" << fBoardID
 		  << ".Channel" << ch << ".MemoryFull"; 
    
+    CAEN_DGTZ_ErrorCode retcod;
 
-    CAEN_DGTZ_ReadTemperature(fHandle, ch, &(ch_temps[ch]));
-    TLOG_ARB(TTEMP,TRACE_NAME) << tempStream.str()
-                               << ": " << ch_temps[ch] << "  C"
-                               << TLOG_ENDL;
-
-    metricMan->sendMetric(tempStream.str(), int(ch_temps[ch]), "C", 11,
-			  artdaq::MetricMode::Average);
-
-    if( ch_temps[ch] > fCAEN.maxTemp ){ // V1730(S) shuts down at 70(85) celsius
-      TLOG(TLVL_ERROR) << "CAENV1730 BoardID " << fBoardID << " : "
-                       << "Temperature above " << fCAEN.maxTemp << " degrees Celsius for channel " << ch
-		       << TLOG_ENDL;
+    if ( fCAEN.temperatureCheckMask & ( 1 << ch ) ) // Channels temperature readout enabled?
+    {
+      retcod = CAEN_DGTZ_ReadTemperature(fHandle, ch, &(ch_temps[ch]));
+      TLOG_ARB(TTEMP,TRACE_NAME) << tempStream.str()
+				 << ": " << ch_temps[ch] << "  C"
+				 << TLOG_ENDL;
+      
+      metricMan->sendMetric(tempStream.str(), int(ch_temps[ch]), "C", 11,
+			    artdaq::MetricMode::Average);
+
+      //Need 3 requirements to shut down for high temperature: 
+      // 1. Can successfully reead temperature: return code = 0 since S/N 164 can fail to read temperature during acquisition
+      // 2. Temperature > Requirement , 70C for V1730 and 85C for V1730S
+      // 3. Temperature < non_physical temperature, 200C since S/N 164 can produce non-physical temperature
```

Comment:
Given my proposed changes, I would reformulate this to better describe the logic below it:
```suggestion
      //Need 3 requirements to shut down for high temperature: 
      // 1. Can successfully read temperature: return code = 0 since S/N 164 can fail to read temperature during acquisition
      // 2. Temperature < non_physical temperature, 200C since S/N 164 can produce non-physical temperature
      // 3. Temperature > Requirement , 70C for V1730 and 85C for V1730S
```